SilKit/source/services/orchestration/TimeSyncService.cpp

```diff
@@ -546,6 +612,75 @@ bool TimeSyncService::AbortHopOnForCoordinatedParticipants() const
     }
     return false;
 }
+
+auto TimeSyncService::GetCurrentWallClockSyncPoint() const -> std::chrono::nanoseconds
+{
+    Lock lock{_mx};
+    return _currentWallClockSyncPoint;
+}
+
+void TimeSyncService::StartWallClockCouplingThread()
+{
+    _wallClockCouplingThreadRunning = true;
+    _stopFuture = _wallClockCouplingThreadStopPromise.get_future();
+
+    _wallClockCouplingThread = std::thread{[&]() {
+
+        SilKit::Util::SetThreadName("SilKit-WallClockCoupling");
+
+        auto startTime = std::chrono::steady_clock::now();
+        const auto duration = _timeConfiguration.NextSimStep().duration;
+        auto nextWallClockSyncPoint = duration;
+
+        while (true)
+        {
+            auto futureStatus = _stopFuture.wait_for(1ms); // Resolution
+
+            if (futureStatus == std::future_status::ready)
+            {
+                // stop was signaled; stopping thread;
+                return;
+            }
+
+            const auto currentRunDuration = std::chrono::steady_clock::now() - startTime;
+            const auto currentVirtualTime = Now();
+
+            if (_isCoupledToWallClock && currentRunDuration / _animationFactor > nextWallClockSyncPoint)
```

Comment:
I'm unsure if this works correctly if the execution isn't continuous, e.g., because the program is interrupted by a debugger.

(written before running / testing / playing with the code)

Reply:
I don't see a good way of excluding the debugger time. I could try to capture debugger events but I guess this is highly non-trivial. Another way would be to implement a kind of additive clock (mytime += elapsed) that would be  halted while debugging. This would be not very accurate I guess.. I could live with the behavior that there is a "catching-up phase" after debugging.